### PR TITLE
[client] Fix macOS DNS panic on malformed scutil output

### PR DESCRIPTION
**Pull request by lixmal**
## Describe your changes

The macOS DNS host manager parsed `scutil show State:/Network/Service/<id>/DNS` with unchecked `strings.Split(line, " : ")[1]`. On hosts where an array entry comes back without a value, that panics with `index out of range [1] with length 1`, taking the daemon down on every network map update. The reporter's host has an orphaned "USB 10/100/1000 LAN" service with a null hardware address that macOS refuses to remove, so the daemon crash-looped on every sync and `--disable-dns` was the only workaround.

Parsing now lives in `parseSystemDNSSettings`, which splits with `SplitN` and skips lines without a value instead of indexing blindly. `getPrimaryService` gets the same treatment for its `PrimaryService` / `Router` lines.

## Issue ticket number and link

Fixes #7173

### Checklist
- [x] Is it a bug fix
- [ ] Is a typo/documentation fix
- [ ] Is a feature enhancement
- [ ] It is a refactor
- [x] Created tests that fail without the change (if possible)
- [x] I ran and tested this change locally — I did not rely on CI to find out whether it works
- [x] This PR has a single purpose (not a fix + refactor + feature in one)
- [x] This change is a trivial fix, **OR** it links an issue the NetBird team agreed on beforehand.

> By submitting this pull request, you confirm that you have read and agree to the terms of the [Contributor License Agreement](https://github.com/netbirdio/netbird/blob/main/CONTRIBUTOR_LICENSE_AGREEMENT.md).

## Documentation
Select exactly one:

- [ ] I added/updated documentation for this change
- [x] Documentation is **not needed** for this change (crash fix in DNS parsing, no user-facing behavior or configuration change)

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved DNS configuration parsing on macOS.
  * Correctly handles IPv4, IPv6, IPv4-mapped IPv6, invalid, empty, and unspecified addresses.
  * Applies the standard DNS port when no port is provided.
  * Prevents malformed entries and empty domains from being included.
* **Tests**
  * Added comprehensive coverage for DNS parsing and edge cases.

### CODE REVIEW

**Review by pappz — APPROVED** [review]

File: client/internal/dns/host_darwin.go [hunk 1]
```diff
@@ -288,36 +308,45 @@ func (s *systemConfigurator) getSystemDNSSettings() (SystemDNSSettings, error) {
...
+
+		ip, err := netip.ParseAddr(value)
+		if err != nil || ip.IsUnspecified() {
+			continue
+		}
+		ip = ip.Unmap()
+		serverAddresses = append(serverAddresses, ip)
+		// Prefer the first IPv4 server as ServerIP since our DNS listener is IPv4.
+		if !dnsSettings.ServerIP.IsValid() && ip.Is4() {
+			dnsSettings.ServerIP = ip
```
[the remark this answers is not in the document]
Comment (lixmal) [reply client/internal/dns/host_darwin.go:341]: This part is just a small refactor, open a gh issue for this please
